src/Language/Haskell/Brittany/Internal/LayouterBasics.hs

```diff
@@ -415,8 +423,24 @@ docExt x anns shouldAddComment = allocateNode $ BDFExternal
 docAlt :: [ToBriDocM BriDocNumbered] -> ToBriDocM BriDocNumbered
 docAlt l = allocateNode . BDFAlt =<< sequence l
 
-docAltFilter :: [(Bool, ToBriDocM BriDocNumbered)] -> ToBriDocM BriDocNumbered
-docAltFilter = docAlt . map snd . filter fst
+newtype CollectAltM a = CollectAltM (Writer.Writer [ToBriDocM BriDocNumbered] a)
+  deriving (Functor, Applicative, Monad)
+
+addAlternativeCondM :: Bool -> CollectAltM (ToBriDocM BriDocNumbered) -> CollectAltM ()
```

Comment:
I don't really see the usecase for this, and one might expect the same pattern as with `when/whenM`:

~~~~.hs
when :: Applicative f => Bool -> f () -> f ()
whenM :: Monad m => m Bool -> m () -> m ()
~~~~

(Also it is not used anywhere afaict.)

Reply:
Indeed it's not used anywhere. I think I tried to use it somewhere but found a way without it.